### PR TITLE
Navigation | New Home tab to priorize all unread - BLOCKED

### PR DESCRIPTION
**Pull request by BillCarsonFr**
App Navigation modification.

This PR introduces a new Home Tab, in this tab you will find all important things you need to read first in a less busy and more compact layout so you can see more at once (one line per room with no last message preview or timestamp) .

It is combining rooms and people into one list, in order to stop people having to change tabs to check on unread rooms/people.

image

The Home tab contains in order
- Invites if any
- Your Favorites for quick access
- DMs with notifications
- Rooms with notifications
- Other Rooms sorted by unread first
- Other Dms
- Suggested rooms of space if any

The existing DM tabs and Room Tabs are modified a bit, they won't show favorite or invites anymore as they will be on Home already. They will continue to display rooms in last activity order with a more detailed view (last message, typing indicator, timestamp, bigger avatar).

- [x] Check Legacy community organization of sections and tabs
- [x] Probably need to update empty state of home
- [ ] Check new user experience, and space joining flow

### CODE REVIEW

**Comment by jwsp1** [comment]
General thought: I don't see the need to separate rooms into categories like Favorites, Unread DM, Unread Rooms, DM, Rooms, and Low Priority. It just wastes valuable space which could be used to show a preview of the latest message. 

If this is supposed to fix #650, then it is not quite what people wanted.
The most simple style for most users would be to remove all tabs and show rooms in this order:
- Favorites (always pinned to top), 
- then DM/Rooms ordered by time of latest message,
- then Low Priority (always pinned to bottom).

**Comment by BillCarsonFr** [comment]
Following feedbacks from this (and related issues), I added a new Preference that allows users to select their preffered layout

image

Pro Layout will have 3 tabs, Home/Dms/Rooms .
Home will put every rooms/dms with notification on top, and will replace the previous optional 3rd tab (notifications only).
Home will have Invites / Favorites and they won't be repeated in other tabs

Simple layout will have no tabs, so all your rooms in one list ordered by activity. 
image

**Review by bmarty — CHANGES_REQUESTED** [review]
First remarks:
- This PR has to be rebased.
- Also please ensure that the sanity test is still passing (it will need some update)
- I will do a full review later

File: vector/src/main/java/im/vector/app/features/home/HomeDetailViewModel.kt [hunk 1]
```diff
@@ -119,6 +122,14 @@ class HomeDetailViewModel @AssistedInject constructor(@Assisted initialState: Ho
...
             HomeDetailAction.MarkAllRoomsRead            -> handleMarkAllRoomsRead()
             is HomeDetailAction.StartCallWithPhoneNumber -> handleStartCallWithPhoneNumber(action)
+            HomeDetailAction.ViewResumed                 -> handleViewResumed()
+        }
+    }
+
+    private fun handleViewResumed() {
+        // Good place to check for layout mode
+        setState {
+            copy(layoutMode = vectorPreferences.getLayoutMode())
```
Comment (bmarty) [inline vector/src/main/java/im/vector/app/features/home/HomeDetailViewModel.kt:132]: setState only if the value is not the same? It's maybe done internally by MvRx, not sure.

File: vector/src/main/java/im/vector/app/features/settings/VectorPreferences.kt [hunk 2]
```diff
@@ -1011,4 +1007,21 @@ class VectorPreferences @Inject constructor(private val context: Context) {
             putInt(TAKE_PHOTO_VIDEO_MODE, mode)
         }
     }
+
+    // Layout
+
+    fun getLayoutMode(): LayoutMode {
+        return defaultPrefs.getString(SETTINGS_LAYOUT_MODE, LayoutMode.PRODUCTIVITY.name).let {
```
Comment (bmarty) [inline vector/src/main/java/im/vector/app/features/settings/VectorPreferences.kt:1014]: Default is PRODUCTIVITY? I would say SIMPLE would be better, or maybe reuse the setting `SETTINGS_LABS_UNREAD_NOTIFICATIONS_AS_TAB`, if true -> PRODUCTIVITY

File: vector/src/main/res/drawable/ic_pro_layout.xml [hunk 3]
```diff
@@ -0,0 +1,84 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (bmarty) [inline vector/src/main/res/drawable/ic_pro_layout.xml:1]: rename the two drawable vectors to ensure they have a common prefix: `ic_layout_pro` and `ic_layout_simple`. Also those are not really icon, so maybe `setting_layout_pro` and `setting_layout_simple` would be better names

File: vector/src/main/res/drawable/ic_simple_layout.xml [hunk 4]
```diff
@@ -0,0 +1,72 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    android:width="121dp"
+    android:height="121dp"
+    android:viewportWidth="121"
+    android:viewportHeight="121">
+  <path
+      android:pathData="M16,17m-8,0a8,8 0,1 1,16 0a8,8 0,1 1,-16 0"
+      android:strokeWidth="1"
+      android:fillColor="#C1C6CD"
```
Comment (bmarty) [inline vector/src/main/res/drawable/ic_simple_layout.xml:9]: does it render well in dark theme?

File: vector/src/main/res/layout/preference_app_choose_layout.xml [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+        <!--            <RadioButton-->
+        <!--                android:id="@+id/proLayoutRadioButton"-->
+        <!--                android:layout_width="0dp"-->
+        <!--                android:layout_height="wrap_content"-->
+        <!--                android:layout_gravity="center_vertical"-->
+        <!--                android:layout_weight="1"-->
+        <!--                android:drawableEnd="@drawable/ic_pro_layout"-->
+        <!--                android:text="@string/settings_home_layout_pro"-->
+        <!--                tools:checked="false" />-->
+
```
Comment (bmarty) [inline vector/src/main/res/layout/preference_app_choose_layout.xml:110]: cleanup?

**Review by bmarty — COMMENTED** [review]
Thanks for doing this, I'm sure it will be really appreciated.
Please see my comments

**Comment by AnisTaluqdar** [comment]
Still, waiting for it. 🥺

**Comment by Flam3z** [comment]
We can only hope and be patient :)

**Comment by kojid0** [comment]
> The merging of this PR has been delayed, the changes it brings require more internal discussions

I understand that the Design Team might wand to unify the look&feel between Android&iOS, but this PR is really necessary for simplicity. 
Currently, you will miss reading messages if you forget to frequently look in both tabs. I have two users where I see that they respond in rooms but not in DMs because they don't check both places. 
Please merge this (as it's a working solution for now) and get back to it later if necessary.

**Comment by bmarty** [comment]
Let's remove this PR, we will iterate with IA later.

**Comment by kojid0** [comment]
Can you please link the subsequent issue / PR? 
Still waiting for this... Many of my friends miss new messages because they don't look in both tabs.

**Pull request closed without merging**